operator/v1/types_machineconfiguration.go

```diff
@@ -158,3 +179,170 @@ const (
+type NodeDisruptionPolicyClusterStatus struct {
+	// files is a list of MachineConfig file definitions and actions to take to changes on those paths
+	// +optional
+	// +listType=map
+	// +listMapKey=path
+	// +kubebuilder:validation:MaxItems=100
+	Files []NodeDisruptionPolicyFile `json:"files,omitempty"`
+	// units is a list MachineConfig unit definitions and actions to take on changes to those services
+	// +optional
+	// +listType=map
+	// +listMapKey=name
+	// +kubebuilder:validation:MaxItems=100
+	Units []NodeDisruptionPolicyUnit `json:"units,omitempty"`
+	// sshkey is the overall sshkey MachineConfig definition
+	// +optional
+	SSHKey NodeDisruptionPolicySSHKey `json:"sshkey,omitempty"`
+}
+
+// NodeDisruptionPolicyFile is a file entry and corresponding actions to take
+type NodeDisruptionPolicyFile struct {
+	// path is the file path to a file on disk managed through a MachineConfig.
+	// Actions specified will be applied when changes to the file at the path
+	// configured in this field.
+	// +kubebuilder:validation:Required
+	Path string `json:"path"`
+	// actions represents the series of commands to be executed on changes to the file at
+	// corresponding file path. This is an atomic list, which will be validated by
+	// the MachineConfigOperator, with any conflicts reflecting as an error in the
+	// status. If validation is successful, the actions will be applied in the order
+	// they are set in the list. If there are other incoming changes to other MachineConfig
+	// entries in the same update that require a reboot, the reboot will supercede these actions.
+	// +kubebuilder:validation:Required
+	// +listType=atomic
+	// +kubebuilder:validation:MaxItems=10
+	// +kubebuilder:validation:XValidation:rule="self.exists(x, x.type=='Reboot') ? size(self) == 1 : true", message="Reboot action can only be specified standalone, as it will override any other actions"
+	// +kubebuilder:validation:XValidation:rule="self.exists(x, x.type=='None') ? size(self) == 1 : true", message="None action can only be specified standalone, as it will override any other actions"
+	Actions []NodeDisruptionPolicyAction `json:"actions"`
+}
+
+// NodeDisruptionPolicyUnit is a systemd unit name and corresponding actions to take
+type NodeDisruptionPolicyUnit struct {
+	// name represents the service name of a systemd service managed through a MachineConfig
+	// Actions specified will be applied for changes to the named service
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:XValidation:rule="self.endsWith('.service') || self.endsWith('.socket') || self.endsWith('.device') || self.endsWith('.mount') || self.endsWith('.automount') || self.endsWith('.swap') || self.endsWith('.target') || self.endsWith('.path') || self.endsWith('.timer') || self.endsWith('.snapshot') || self.endsWith('.slice') || self.endsWith('.scope')",message="Invalid service name schema, needs to be in the format of ${NAME}.${SERVICETYPE}"
```

Comment:
if name should be more than zero chars, need another CEL rule in addition to this one.